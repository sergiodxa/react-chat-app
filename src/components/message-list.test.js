import React from "react";
import { render } from "@testing-library/react";

import MessageList from "./message-list";

test("ChannelList component", () => {
  const messages = [
    {
      content: "hola",
      username: "sergiodxa",
      date: "2019-06-02T00:45:43.963Z",
      channel: "general"
    },
    {
      content: "hola, que tal?",
      username: "sergiodxa",
      date: "2019-06-01T00:45:43.963Z",
      channel: "general"
    },
    {
      content: "hola, que tal?",
      username: "sergiodxa",
      date: "2019-05-03T00:45:43.963Z",
      channel: "general"
    }
  ];

  const { asFragment } = render(
    <MessageList messages={messages} now={new Date()} />
  );

  expect(asFragment()).toMatchSnapshot();
});
