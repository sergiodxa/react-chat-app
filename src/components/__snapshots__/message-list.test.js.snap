// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelList component 1`] = `
<DocumentFragment>
  .emotion-3 {
  padding: 1rem 1.5rem;
  border-bottom: 1px solid #eaeaea;
}

.emotion-3:last-child {
  border-bottom: none;
}

.emotion-0 {
  font-size: 1.1rem;
}

.emotion-1 {
  font-size: .75rem;
  font-style: normal;
  color: #666;
}

.emotion-2 {
  margin: .5rem 0 0;
}

<section>
    <article
      class="emotion-3"
    >
      <div>
        <strong
          class="emotion-0"
        >
          sergiodxa
        </strong>
         
        <time
          class="emotion-1"
        >
          19:45:43
        </time>
      </div>
      <p
        class="emotion-2"
      >
        hola
      </p>
    </article>
    <article
      class="emotion-3"
    >
      <div>
        <strong
          class="emotion-0"
        >
          sergiodxa
        </strong>
         
        <time
          class="emotion-1"
        >
          19:45:43
        </time>
      </div>
      <p
        class="emotion-2"
      >
        hola, que tal?
      </p>
    </article>
    <article
      class="emotion-3"
    >
      <div>
        <strong
          class="emotion-0"
        >
          sergiodxa
        </strong>
         
        <time
          class="emotion-1"
        >
          19:45:43
        </time>
      </div>
      <p
        class="emotion-2"
      >
        hola, que tal?
      </p>
    </article>
  </section>
</DocumentFragment>
`;
